django/.coveragerc

```diff
@@ -22,6 +22,8 @@ exclude_lines =
     raise AssertionError
     raise NotImplementedError
     if __name__ == .__main__.:
+    @abstractmethod
+    @abc.abstractmethod
```

Comment:
Csn abstract methods hold logic? If yes, probably shouldn't be ignored from codecov

Reply:
> Csn abstract methods hold logic? If yes, probably shouldn't be ignored from codecov. (feel free to resolve this to unblock merge after replying, not intending to keep as a blocker)

@mythic Here's [a discussion](https://github.com/pytest-dev/pytest-cov/issues/428) I read before adding the excludes. TL;DR:

* Abstract methods CAN hold logic that's intended to be called via `super` in subclasses
* Abstract methods can't be tested directly; indirectly they can be tested by testing a method in a concrete subclass that calls `super`'s implementation
* In case of [abstract properties](https://github.com/thunderstore-io/Thunderstore/pull/567/commits/f0804858c160d7b0b3055ffa71b482e9fc65ee0d#diff-c89dcd20bd5260885b9de41286da82915e2966d84ea99a67b39aec4e43e0179dR11-R14) that are implemented as [static values](https://github.com/thunderstore-io/Thunderstore/pull/567/commits/f0804858c160d7b0b3055ffa71b482e9fc65ee0d#diff-c89dcd20bd5260885b9de41286da82915e2966d84ea99a67b39aec4e43e0179dR79), calling `super` really makes no sense, so abstract properties can't be tested properly this way
* Alternatively they could be excluded by using regexp to exclude all functions that contain only `pass`, but I thought that would be problematic, since it would also apply to non-abstract functions, which can be a code smell in and of themselves. It could be nice if test coverage would bring attention to them
* Yet another alternative would be that the abstract properties only contain a docstring, and no body nor `pass`, but this would prolly mean adding unnecessary docstrings, which feels dirty and I think this sould be solved via configuration instead
* Not testing nor excluding the abstract properties seems like a poor idea too, since CI pipeline is adamant about not lowering the coverage percentage, but in a small file a few abstract properties can cause a serious drop in the percentage.

Thoughts?

Reply:
> I wonder if implementing a stub subclass with super calls within a test would be the least effort all around,

That might be practical, but IMO ideologically so wrong. Test suite and coverage report exist to serve us. I don't think we should start jumping through its hoops, writing tests we know to be pointless, like we're trying to appease some mad god (in b4 @Oksamies and Warhammer 40k).

> I'm not opposed to using some regex magic to exclude empty abstract methods from test coverage

AFAICT there's no way to limit it only to abstract methods. Now that I think of it, it would probably also exclude other places where `pass` might be used and thus has it's own problems.

---

As a compromise between pragmatic and ideological, I ended up removing `pass`es and adding docstrings to abstract properties/methods. They are now ignored by the coverage without having to edit the exclude config. If anyone adds some content to them in the future, it will decrease the coverage, and hopefully nudge that person to add some tests.